api/functions/src/downloads/server.ts

```diff
@@ -8,8 +8,12 @@ export function ConstructFileDownloadServer() {
   app.options('*');
 
   app.get('/machine/:machineId/sensor/:sensorId', async (req, res) => {
-    if (!req.params.sensorId || !req.params.machineId) {
-      res.status(400).send('Bad Request');
+    if (!req.params.sensorId) {
+      res.status(400).send('No sensorId provided');
+      return;
+    }
+    if (!req.params.machineId) {
+      res.status(400).send('No machineId provided');
```

Comment:
Technically this addition shouldn't be in this PR but it doesn't matter - I'm just separating out the checks and sending more helpful error messages

Reply:
Good spotting, that existing check wasn't helpful for debugging.